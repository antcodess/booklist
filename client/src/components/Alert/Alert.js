import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectAlertItem } from '../../redux/alert/alert.selectors';

import './Alert.scss';

export const Alert = ({ alert }) => {

  return (
    <React.Fragment>
      { 
        alert ?
          <section className='alert-container'>
            <div className={`alert ${alert.status}`}>
              <div className='alert-content'>
                <span className='alert-message'>
                  {alert.message}
                </span>
              </div>
            </div>
          </section>
        : <section className='alert-container'></section>
      }
    </React.Fragment>
  );
};

const mapStateToProps = createStructuredSelector({
  alert: selectAlertItem
});

export default connect(mapStateToProps)(Alert);