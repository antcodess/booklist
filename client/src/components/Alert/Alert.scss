.alert-container {
  position: relative;
  
  .alert {
    position: fixed;
    width: 100%;
    top: 0;
    z-index: 8;
    padding: 3px 0px;

    &.success {
      background-color: #2ab500;
    }

    &.failure {
      background-color: red;
    }

    &.regular {
      background-color: #0080DE;
    }

    .alert-content {
      margin: auto;
      display: flex;
      justify-content: center;
      margin-top: 5px;

      .alert-message {
        font-family: "Hurme Geometric Sans 3 W03 Bold";
        color: white;
        text-align: center;
        padding-bottom: 5px;
      }
    }
  }
}