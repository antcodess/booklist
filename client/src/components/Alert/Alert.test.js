import React from 'react';
import { shallow } from 'enzyme';
import { Alert } from './Alert';


describe('Alert Component', () => {
  it('should render Alert Component', () => {
    const mockAlert = {
      status: 'success',
      message: 'This is an alert'
    };
    const wrapper = shallow(<Alert alert={mockAlert} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not show alert if alert is null', () => {
    const mockAlert = null;
    const wrapper = shallow(<Alert alert={mockAlert} />);
    expect(wrapper.find('section[className="alert-container"]').children().length).toEqual(0);
  });

  it('should show alert if alert has value', () => {
    const mockAlert = {
      status: 'success',
      message: 'This is an alert'
    };
    const wrapper = shallow(<Alert alert={mockAlert} />);
    expect(wrapper.find('section[className="alert-container"]').children().length).toEqual(1);
  });
});
