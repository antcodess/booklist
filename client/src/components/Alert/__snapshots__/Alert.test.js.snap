// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component should render Alert Component 1`] = `ShallowWrapper {}`;
